### DISCUSSION TITLE
multiple middleware like laravel

### DISCUSSION THREAD

**Discussion by sirajyesuf** [nutgram/nutgram#140] [Q&A] [answered]
hello guys here is my question when I apply multiple middlewares on  my handler only the last one execute 
like this 
$bot->onCommand('start {parameter}', ClaimHandler::class)
    ->middleware(Authenticate::class)
    ->middleware(Approved::class)
    ->middleware(ClaimProduct::class);
    the last middleware claimproduct only run.
    i want to apply multiple middleware on the handler  pls help me how i can do this?

**Comment by Lukasss93** [comment]
Hello!
Your middleware class must have `$next($bot);`

Example:
```php
namespace App\Telegram\Middleware;

use SergiX44\Nutgram\Nutgram;

class OnlyDev
{
    public function __invoke(Nutgram $bot, $next): void
    {
        if ($bot->user()?->id !== config('developer.id')) {
            return;
        }

        $next($bot);
    }
}
```

Docs: https://nutgram.dev/docs/usage/middleware

**Comment by sergix44** [comment, marked as answer]
In addition, be aware that right now the middlewares are a LIFO queue, so in your example, the middlewares are executed in this order:
```
ClaimProduct::class => Approved::class => Authenticate::class
```
Maybe is not superclear from the docs
